### PR TITLE
Sdil 941 tax enrolment

### PR DESCRIPTION
**Pull request by mattrobertsky**
please review.

### CODE REVIEW

File: app/uk/gov/hmrc/softdrinksindustrylevy/connectors/TaxEnrolmentConnector.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class TaxEnrolmentConnector extends ServicesConfig {
+
+  val callbackUrl: String = getConfString("tax-enrolments.callback", "")
+  val serviceName: String = getConfString("tax-enrolments.serviceName", "")
+
+  val http: WSHttp.type = WSHttp
```
Comment (richard-dennehy) [inline app/uk/gov/hmrc/softdrinksindustrylevy/connectors/TaxEnrolmentConnector.scala]: can this be inlined if it's just going to be a fixed reference to an Object that can't be overridden?

File: app/uk/gov/hmrc/softdrinksindustrylevy/models/RosmRegisterRequest.scala [hunk 2]
```diff
@@ -19,7 +19,7 @@ package uk.gov.hmrc.softdrinksindustrylevy.models
 import play.api.libs.json.Json
 
 case class RosmRegisterRequest(
-                                regime: String = "ZSDL",
+                                regime: String = "ZSDL", // TODO turn this to a config
```
Comment (richard-dennehy) [inline app/uk/gov/hmrc/softdrinksindustrylevy/models/RosmRegisterRequest.scala]: do

File: conf/application.conf [hunk 3]
```diff
@@ -152,6 +158,14 @@ microservice {
       token = "senbuijkfeshnoiufesnloijfesnoifes"
       environment = "live"
     }
+
+    tax-enrolments {
+      host = localhost
+      port = 9995 // TODO get Adam to figure out what this is used for
+      serviceName = "ZSDL"
+      // TODO update the callback
+      callback = "http://localhost:8701/tax-enrolment-callback/"
```
Comment (richard-dennehy) [inline conf/application.conf]: do

File: conf/application.conf [hunk 4]
```diff
@@ -152,6 +158,14 @@ microservice {
       token = "senbuijkfeshnoiufesnloijfesnoifes"
       environment = "live"
     }
+
+    tax-enrolments {
+      host = localhost
+      port = 9995 // TODO get Adam to figure out what this is used for
```
Comment (richard-dennehy) [inline conf/application.conf]: do/remove

File: app/uk/gov/hmrc/softdrinksindustrylevy/connectors/TaxEnrolmentConnector.scala [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      }
+    }
+  }
+
+  private def subscribeUrl(subscriptionId: String) =
+    s"${baseUrl("tax-enrolments")}/tax-enrolments/subscriptions/$subscriptionId/subscriber"
+
+  private def requestBody(safeId: String, formBundleNumber: String): JsObject = {
+    Json.obj(
+      "serviceName" → JsString(serviceName),
```
Comment (richard-dennehy) [inline app/uk/gov/hmrc/softdrinksindustrylevy/connectors/TaxEnrolmentConnector.scala]: remove unnecessary JsValue wrappers pls

File: app/uk/gov/hmrc/softdrinksindustrylevy/controllers/SdilController.scala [hunk 6]
```diff
@@ -32,14 +32,16 @@ import scala.concurrent.ExecutionContext.Implicits.global
...
 @Singleton
 class SdilController @Inject()(desSubmissionService: DesSubmissionService,
-                               desConnector: DesConnector,
+															 taxEnrolmentConnector: TaxEnrolmentConnector,
+															 desConnector: DesConnector,
                                mongo: MongoStorageService) extends BaseController {
 
-  def submitRegistration(idType: String, idNumber: String): Action[JsValue] = Action.async(parse.json) { implicit request =>
-    withJsonBody[Subscription](data =>
+	def submitRegistration(idType: String, idNumber: String, safeId: String): Action[JsValue] = Action.async(parse.json)  { implicit request =>
```
Comment (richard-dennehy) [inline app/uk/gov/hmrc/softdrinksindustrylevy/controllers/SdilController.scala]: indenting with tabs 😞

File: app/uk/gov/hmrc/softdrinksindustrylevy/connectors/TaxEnrolmentConnector.scala [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+  val callbackUrl: String = getConfString("tax-enrolments.callback", "")
+  val serviceName: String = getConfString("tax-enrolments.serviceName", "")
+
+  val http: WSHttp.type = WSHttp
+
+  def subscribe(safeId: String, formBundleNumber: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] = {
+    http.PUT[JsObject, HttpResponse](subscribeUrl(formBundleNumber), requestBody(safeId, formBundleNumber)) map {response =>
+      response.status match {
+        case 400 | 401 => {
```
Comment (nkaraolis) [inline app/uk/gov/hmrc/softdrinksindustrylevy/connectors/TaxEnrolmentConnector.scala]: This case won't get hit as http verbs returns an exception for 4XX and 5XX status codes so would need to be in a recover case

**Pull request merged**